### DISCUSSION TITLE
Siemens Biograph mCT list mode 32bits .ptd

### DISCUSSION THREAD

**Discussion by wanbint** [villekf/OMEGA#44] [Q&A] [answered]
Hi, 

I am wondering if anybody knows how to read the Siemens Biograph mCT list mode 32bits ended with .ptd. I see there is a main file _Biograph_mCT_main.m_ in the code. I tried both **options.use_machine** = 2 and 3, but neither worked on my case.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%% BIOGRAPH mCT DATA %%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% Use Biograph mCT data
% 0/false = Don't use Biograph data (use GATE data instead)
% 1 = Use 64-bit list-mode data
% 2 = Use sinogram data (.s/.ptd)
% 3 = Use 32-bit list-mode data
options.use_machine = 3;

**Comment by villekf** [comment]
Hi,

3 should, theoretically, work. However, I haven't actually been able to test .ptd-files with mCT so that feature was never tested. What issues are you running into?

**Comment by wanbint** [comment]
Hi, thanks for replying to me.

I was running **Biograph_mCT_main.m** with the **options.use_machine = 3**. When it runs into **load_data_mCT.m**, the Matlab crashed with the following error:
![image](https://github.com/villekf/OMEGA/assets/72076002/2933e0a0-6162-4b9e-b0e7-a925a5c0db1f)

The error is coming from list2sinogram.mexw64, and I cannot debug it in Matlab since it is a built MEX function.

I am also attaching an example .ptd file from a Siemens Biograph mCT for your testing. 
https://drive.google.com/file/d/12O2qJhFdafSVUqHfTStjl5hWyWw86tgU/view?usp=sharing

best,

**Comment by villekf** [comment]
The reason is probably that as I never got 32-bit listmode data as a ptd-files, which are slightly different from the "regular" listmode files, the current code most likely can't handle the extra header data in the ptd-files. Thanks for the file though, I'll try to see if I can get it working sometime next week.

**Reply by wanbint** [reply]
Hi, thanks for testing. 
If I understand it correctly, the ptd-file includes the extra header data at the end of the ptd-file. Here is part of the header file that I tried to read from the shared ptd-file.

% ***************************** header ***************************** %
PET data type:=Emission
data format:=CoincidenceList
horizontal bed translation:=stepped
start horizontal bed position (mm):=-1115.01
end horizontal bed position (mm):=-1115.01
start vertical bed position (mm):=125.5
%bed zero offset (mm):=-512.5
number of energy windows:=1
%energy window lower level (keV) [1]:=435
%energy window upper level (keV) [1]:=650

!PET STUDY (Emission data):=
PET scanner type:=cylindrical
transaxial FOV diameter (cm):=70.4
number of rings:=55
distance between rings (cm):=0.4054
gantry tilt angle (degrees):=0
gantry crystal radius (cm):=42.76
bin size (cm):=0.2005
septa state:=none
%number of TOF time bins:=13
%TOF mashing factor:=1

!IMAGE DATA DESCRIPTION:=
%preset type:=time
%preset value:=240
%preset unit:=seconds
image duration (sec):=240
%total listmode word counts:=50402066

%COINCIDENCE LIST DATA:=
%LM event and tag words format (bits):=32
%timing tagwords interval (msec):=1
%singles polling method:=instantaneous
%singles polling interval (sec):=2
%singles scale factor:=8
%total number of singles blocks:=48
%axial compression:=11
%maximum ring difference:=49
%number of projections:=400
%number of views:=168
%number of segments:=9
%segment table:={109,97,97,75,75,53,53,31,31}
%comment:=PET/CT gantry offset during PET acquisition was x=0.000000mm, y=0.000000mm, z=0.000000mm
% ***************************** header ***************************** %

Hope it helps. Let me know if you need anything else from me.

Thanks,

**Reply by villekf** [reply, marked as answer]
I updated the [devel](https://github.com/villekf/OMEGA/blob/devel/biograph/list2sinogram.mexw64) version first in case things get messed up. On my end it is working fine, though you seem to be using larger mashing value so be sure to half the default number of views (angles). Let me know in case it works or doesn't work. I compiled it with VS 2022 and Matlab 2021b so earlier Matlab versions could potentially have issues with it.

**Comment by wanbint** [comment]
I tested the new devel version and it worked pretty well on my computer. Thanks for updating the devel version and I appreciate your work.
